hazelcast/src/main/java/com/hazelcast/collection/impl/list/ListContainer.java

```diff
@@ -166,7 +166,9 @@ public int indexOf(boolean last, Data value) {
         if (itemList == null) {
             if (itemMap != null && !itemMap.isEmpty()) {
                 itemList = new ArrayList<CollectionItem>(itemMap.values());
-                Collections.sort(itemList);
+                sort(itemList);
+                CollectionItem lastItem = itemList.get(itemList.size() - 1);
+                setId(lastItem.getItemId() + ID_PROMOTION_OFFSET);
```

Comment:
We enter this branch, when backups of `ILIst` items are promoted. The backups are stored in the `itemMap` and are used to populate the `itemList`. The issue was to not call `setId()` with the maximum `itemId` of the existing items.

Since the `itemList` is already sorted (by the `itemId` its `CollectionItems`), we can just pick the `lastItem` to retrieve the maximum `itemId`.